benchmark/single-source/Substring.swift

```diff
@@ -267,6 +268,38 @@ public func run_SubstringComparable(_ N: Int) {
   CheckResults(count == N*500)
 }
 
+extension Character {
+  fileprivate var isASCIIWhitespace: Bool {
+    return self == " " || self == "\t" || self == "\r" || self == "\n" || self == "\r\n"
+  }
+}
+
+extension Substring {
+  fileprivate func trimWhitespace() -> Substring {
+    var me = self
+    while me.first?.isASCIIWhitespace == .some(true) {
+      me = me.dropFirst()
+    }
+    while me.last?.isASCIIWhitespace == .some(true) {
+      me = me.dropLast()
+    }
+    return me
+  }
+}
+
+let _trimmableSubstrings = "pineapple,🍍,  pineapple\t,\r\n\r\n\r\n, 🍍 ,".split(separator: ",")
```

Comment:
Note to future self for a follow-up PR: Make this dynamically generated during setup. We don't want to allow for any constant folding and we do want to run over ARC-managed storage.